tests/e2e/tidbcluster/tidbcluster.go

```diff
@@ -287,6 +293,150 @@ var _ = ginkgo.Describe("[tidb-operator] TiDBCluster", func() {
 		oa.StopInsertDataTo(&clusterA)
 	})
 
+	ginkgo.It("Adhoc backup and restore with BR CRD", func() {
+		if framework.TestContext.Provider != "aws" {
+			framework.Skipf("provider is not aws, skipping")
+		}
+		tcNameFrom := "backup"
+		tcNameTo := "restore"
+		serviceAccountName := "tidb-backup-manager"
+		backupFolder := time.Now().Format(time.RFC3339)
+
+		// create backup cluster
+		tcFrom := fixture.GetTidbCluster(ns, tcNameFrom, "v4.0.0-beta.1")
+		tcFrom.Spec.PD.Replicas = 1
+		tcFrom.Spec.TiKV.Replicas = 1
+		tcFrom.Spec.TiDB.Replicas = 1
+		err := genericCli.Create(context.TODO(), tcFrom)
+		framework.ExpectNoError(err)
+		err = oa.WaitForTidbClusterReady(tcFrom, 30*time.Minute, 15*time.Second)
+		framework.ExpectNoError(err)
+		clusterFrom := newTidbClusterConfig(e2econfig.TestConfig, ns, tcNameFrom, "", "")
+
+		// create restore cluster
+		tcTo := fixture.GetTidbCluster(ns, tcNameTo, "v4.0.0-beta.1")
+		tcTo.Spec.PD.Replicas = 1
+		tcTo.Spec.TiKV.Replicas = 1
+		tcTo.Spec.TiDB.Replicas = 1
+		err = genericCli.Create(context.TODO(), tcTo)
+		framework.ExpectNoError(err)
+		err = oa.WaitForTidbClusterReady(tcTo, 30*time.Minute, 15*time.Second)
+		framework.ExpectNoError(err)
+		clusterTo := newTidbClusterConfig(e2econfig.TestConfig, ns, tcNameTo, "", "")
+
+		// import some data to sql with blockwriter
+		ginkgo.By(fmt.Sprintf("Begin inserting data into cluster %q", clusterFrom.ClusterName))
+		oa.BeginInsertDataToOrDie(&clusterFrom)
+		err = wait.PollImmediate(time.Second*5, time.Minute*5, utiltidb.TiDBIsInserted(fw, tcFrom.GetNamespace(), tcFrom.GetName(), "root", "", "test", "block_writer"))
+		framework.ExpectNoError(err)
+		ginkgo.By(fmt.Sprintf("Stop inserting data into cluster %q", clusterTo.ClusterName))
```

Comment:
```suggestion
		ginkgo.By(fmt.Sprintf("Stop inserting data into cluster %q", clusterFrom.ClusterName))
```

it's `clusterFrom` you are inserting data into.

Reply:
done